File: static/java/concurrency/SimpleWebServer.java
// @include
  public class SimpleWebServer {
    final static int PORT = 8080;
    public static void main (String [] args) throws IOException {
      ServerSocket serversock = new ServerSocket(PORT);
      for (;;) {
        Socket sock = serversock.accept();
        ProcessReq(sock);
      }
    }
  }
// @exclude

File: solutions/python/simple_web_server.py
import socket


def process_req(sock):
    while True:
        data = sock.recv(1024)
        if not data:
            break
        print(data)
        sock.sendall(data)


# @include
SERVERPORT = 8080


def main():
    serversock = socket.socket(socket.AF_INET, socket.SOCK_STREAM)
    serversock.bind(('', SERVERPORT))
    serversock.listen(5)
    while True:
        sock, addr = serversock.accept()
        process_req(sock)
# @exclude

if __name__ == '__main__':
    main()
